packages/grid-pro/test/keyboard-navigation.test.js

```diff
@@ -0,0 +1,308 @@
+      const firstCell = getContainerCell(grid.$.items, 1, 1);
+      dblclick(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      const secondCell = getContainerCell(grid.$.items, 1, 0);
+      const spy = sinon.spy(secondCell, 'focus');
+      tab(input, ['shift']);
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should focus cell next available for editing on the next row in non-edit mode on Tab', () => {
+      const firstCell = getContainerCell(grid.$.items, 1, 1);
+      dblclick(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      const secondCell = getContainerCell(grid.$.items, 2, 0);
+      const spy = sinon.spy(secondCell, 'focus');
+      tab(input);
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should focus previous cell available for editing on the previous in non-edit mode on Shift Tab', () => {
+      const firstCell = getContainerCell(grid.$.items, 2, 0);
+      dblclick(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      const secondCell = getContainerCell(grid.$.items, 1, 1);
+      const spy = sinon.spy(secondCell, 'focus');
+      tab(input, ['shift']);
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should focus editable cell on the next row in non-edit mode on Enter, if `enterNextRow` is true', () => {
+      grid.enterNextRow = true;
+      const firstCell = getContainerCell(grid.$.items, 1, 0);
+      enter(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      const secondCell = getContainerCell(grid.$.items, 2, 0);
+      const spy = sinon.spy(secondCell, 'focus');
+      enter(input);
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should exit the edit mode for the cell on Enter, if `enterNextRow` is false', () => {
+      const firstCell = getContainerCell(grid.$.items, 1, 0);
+      dblclick(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      enter(input);
+      expect(getCellEditor(firstCell)).to.be.not.ok;
+    });
+
+    it('should focus editable cell on the previous row in non-edit mode on Shift Enter, if `enterNextRow` is true', () => {
+      grid.enterNextRow = true;
+      const firstCell = getContainerCell(grid.$.items, 1, 0);
+      enter(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      const secondCell = getContainerCell(grid.$.items, 0, 0);
+      const spy = sinon.spy(secondCell, 'focus');
+      enter(input, ['shift']);
+      expect(spy.calledOnce).to.be.true;
+    });
+
+    it('should exit the edit mode for the cell on Shift Enter, if `enterNextRow` is false', () => {
+      const firstCell = getContainerCell(grid.$.items, 1, 0);
+      dblclick(firstCell._content);
+      input = getCellEditor(firstCell);
+
+      enter(input, ['shift']);
+      expect(getCellEditor(firstCell)).to.be.not.ok;
+    });
+
+    it('should focus correct editable cell after column reordering', () => {
```

Comment:
Moved this test inside of the `singleCellEdit` suite and removed setting it in the test.